test/versioned/langchain/tools.tap.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper')
+// load the assertSegments assertion
+require('../../lib/metrics_helper')
+const fs = require('fs')
+// have to read and not require because openai does not export the package.json
+const { version: pkgVersion } = JSON.parse(
+  fs.readFileSync(`${__dirname}/node_modules/@langchain/core/package.json`)
+)
+const { DESTINATIONS } = require('../../../lib/config/attribute-filter')
+const config = {
+  ai_monitoring: {
+    enabled: true
+  },
+  feature_flag: {
+    langchain_instrumentation: true
+  }
+}
+
+tap.test('Langchain instrumentation - tools', (t) => {
+  t.autoend()
+
+  t.before(() => {
+    t.context.agent = helper.instrumentMockedAgent(config)
+    const { WikipediaQueryRun } = require('@langchain/community/tools/wikipedia_query_run')
+    t.context.tool = new WikipediaQueryRun({
+      topKResults: 3,
+      maxDocContentLength: 4000
+    })
+  })
+
+  t.afterEach(() => {
+    t.context.agent.customEventAggregator.clear()
+  })
+
+  t.teardown(() => {
+    helper.unloadAgent(t.context.agent)
+  })
```

Comment:
This will only be run after all of the subtests have completed. Are you sure we don't want to clear the agent before every subtest? As it is, each subtest is going to retain state from which ever test ran before it.

Reply:
ok so now i remember. i was just following what we did in [openai](https://github.com/newrelic/node-newrelic/blob/main/test/versioned/openai/chat-completions.tap.js).